### PR TITLE
Build once, test many

### PR DESCRIPTION
**Pull request by emlowe**
Trying to improve the build/test workflow 

Since this is using ABI3, we only need to build the wheel once time and then test it using a python matrix

I think this is ready for merging. Further work can happen on top of this. Still uses ABI3

### CODE REVIEW

**Comment by prozacchiwawa** [comment]
Thanks

File: .github/workflows/build-arm64-wheels.yml [hunk 1]
```diff
@@ -23,99 +23,117 @@ jobs:
...
+    container:
+      image: ghcr.io/chia-network/build-images/centos-pypa-rust-aarch64:latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+      - name: Install rust
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          components: rustfmt, clippy
```
Comment (cmmarslender) [inline .github/workflows/build-arm64-wheels.yml:71]: afaik the point of the image you're using (centos-pypa-rust-aarch64) is to have rust pre-bundled, so this step is probably not required.

File: .github/workflows/build-arm64-wheels.yml [hunk 2]
```diff
@@ -23,99 +23,117 @@ jobs:
...
+          rm -rf venv
+          rm -rf dist
+
+  cargo_test:
+    name: Run cargo test on ARM64 Linux
+    runs-on: [ARM64, Linux]
+    container:
+      image: ghcr.io/chia-network/build-images/centos-pypa-rust-aarch64:latest
+    steps:
+      - uses: actions/checkout@v3
```
Comment (cmmarslender) [inline .github/workflows/build-arm64-wheels.yml:65]:
```suggestion
      - uses: actions/checkout@v4
```
v4 is current version of this. would be nice to update while we're in here doing a bunch of changes anyways.

**Comment by emlowe** [comment]
Closing and using Kyle's work going forward

**Pull request closed without merging**